pallets/oracle-collection/src/lib.rs

```diff
@@ -518,26 +693,60 @@ pub mod types {
 
 	pub type OracleValuePair<T> = (<T as Config>::OracleValue, <T as Config>::Timestamp);
 
+	#[derive(Encode, Decode, Clone, PartialEq, Eq, TypeInfo, RuntimeDebug, MaxEncodedLen)]
+	#[scale_info(skip_type_params(T))]
+	pub struct FeederInfo<T: Config> {
+		pub id: T::FeederId,
+		pub max_age: Option<T::Timestamp>,
+		relevance: Relevance,
+	}
+
+	impl<T: Config> FeederInfo<T> {
+		pub fn valid_feed(&self, age_of_feed: T::Timestamp) -> DispatchResult {
+			let too_old = sp_std::cmp::Ordering::Greater == Some(age_of_feed).cmp(&self.max_age);
+
+			if too_old {
+				ensure!(self.dropable(), Error::<T>::FeederMustBeInFeed);
+			}
+
+			Ok(())
+		}
+
+		pub fn dropable(&self) -> bool {
+			self.relevance == Relevance::Dropable
+		}
+	}
+
+	#[derive(Encode, Decode, Clone, PartialEq, Eq, TypeInfo, RuntimeDebug, MaxEncodedLen)]
+	pub enum Relevance {
+		Essential,
+		Neutral,
+		Dropable,
+	}
+
 	/// Type containing the associated info to a key
 	#[derive(Encode, Decode, Clone, PartialEq, Eq, TypeInfo, RuntimeDebug, MaxEncodedLen)]
 	#[scale_info(skip_type_params(T))]
 	pub struct KeyInfo<T: Config> {
-		pub feeders: BoundedBTreeSet<T::FeederId, T::MaxFeedersPerKey>,
+		pub feeders: BoundedBTreeSet<FeederInfo<T>, T::MaxFeedersPerKey>,
```

Comment:
We can have duplicated `FeederId` with different timestamps for different `Relevance`. Are we ok with it?